WordPress/Classes/Utility/Migration/ContentMigrationCoordinator.swift

```diff
@@ -0,0 +1,105 @@
+/// Encapsulates logic related to content migration from WordPress to Jetpack.
+///
+class ContentMigrationCoordinator {
+
+    static let shared: ContentMigrationCoordinator = .init()
+
+    // MARK: Dependencies
+
+    private let dataMigrator: ContentDataMigrating
+    private let keyValueDatabase: KeyValueDatabase
+    private let eligibilityProvider: ContentMigrationEligibilityProvider
+
+    init(dataMigrator: ContentDataMigrating = DataMigrator(),
+         keyValueDatabase: KeyValueDatabase = UserDefaults.standard,
```

Comment:
Can we use the `UserPersistentRepository` instead? It is a more complete form of the `KeyValueDatabase` I wrote to abstract UserDefaults in the project. It contains other functions from the UserDefaults that one might need as well rather than this smaller subset.

Also we can still utilize the `UserPersistentStoreFactory` to keep the codebase uniform. When/if we decide to remove it, and possibly replace it with something else, it would be easier to search for the same thing across the app. Although we are not using it right now really, it is still an abstraction layer that might come handy for the same reason it was needed before imo.

Reply:
> Can we use the UserPersistentRepository instead?

I wanted to use this class, but I figured I had to create a mock implementation on the unit test side — that's why I decided on using the `KeyValueDatabase` instead since it's got an `EphemeralKeyValueDatabase` on the test target.

But then I figured out that there's `InMemoryUserDefaults` that conforms to `UserPersistentRepository` 😅 . So yeah, I can change it.